curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/6350854411ffb73feb6bb84e.md

```diff
@@ -0,0 +1,123 @@
+---
+id: 6350854411ffb73feb6bb84e
+title: Step 12
+challengeType: 0
+dashedName: step-12
+---
+
+# --description--
+
+The `.reduce()` method takes a second parameter that is used as the initial value of the accumulator. To be safe, set your initial value to `0`. Here is an example of setting the initial value to an empty string:
+
+```js
+array.reduce((acc, el) => acc + el.toLowerCase(), "");
+```
```

Comment:
This might be a good chance to cover what will happen if an initial value isn't set:

```suggestion
The `.reduce()` method takes a second parameter that is used as the initial value of the accumulator. Without a second parameter, the `.reduce()` method uses the first element of the array as the accumulator, which can lead to unexpected results.

To be safe, it's best to set an initial value. Here is an example of setting the initial value to an empty string:

```js
array.reduce((acc, el) => acc + el.toLowerCase(), "");
```

Set the initial value of the accumulator to `0`.
```